### PR TITLE
add feature CLIPlugin CR to core-management-plugins package

### PR DESCRIPTION
**Pull request by yharish991**
Signed-off-by: Harish Yayi <yharish991@gmail.com>

### What this PR does / why we need it

This PR makes the feature plugin a context-aware plugin by adding CLIPlugin CR to core-management-plugins package

### Which issue(s) this PR fixes
Fixes https://github.com/vmware-tanzu/tanzu-framework/issues/1618

### Describe testing done for PR

Run `PACKAGE_NAME=featuregates PACKAGE_REPOSITORY=management make package-bundle` and verify the addition of feature CLIPlugin cr in the generated package bundle

### Release note

```release-note
Added feature plugin as a context-aware plugin
```

### PR Checklist

- [x] Squash the commits into one or a small number of logical commits
      
- [x] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [x] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

File: packages/management/core-management-plugins/bundle/config/feature.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+#@ load("@ytt:data", "data")
```
Comment (rajathagasthya) [inline packages/management/core-management-plugins/bundle/config/feature.yaml:1]: The [other files](https://github.com/vmware-tanzu/tanzu-framework/tree/main/packages/management/core-management-plugins/bundle/config) in this directory have a numerical prefix (probably ordering). Should this also follow the same convention?

**Review by rajathagasthya — COMMENTED** [review]
LGTM with minor comments.

> This PR makes the feature plugin a context-aware plugin by adding CLIPlugin CR to core-management-plugins package

Could you please add this description to the commit message body? Also, please fill the section on how this was tested.

File: packages/management/core-management-plugins/bundle/config/feature.yaml (hunk 1 above)
Comment (yharish991) [reply packages/management/core-management-plugins/bundle/config/feature.yaml:1]: numerical prefix doesn't change the order of creation of resources in the package, ordering is done using ordering annotations https://carvel.dev/kapp/docs/v0.45.0/apply-ordering/, ill make another commit to this pr to remove the prefix of the other files

**Review by rajathagasthya — APPROVED** [review]

**Review by anujc25 — APPROVED** [review]
LGTM

**Pull request merged**